x-pack/plugins/apm/public/components/app/correlations/context_popover/field_stats_popover.tsx

```diff
@@ -0,0 +1,163 @@
+import type { FieldTopValuesBucketParams } from '@kbn/unified-field-list-plugin/public';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { useFetchParams } from '../use_fetch_params';
+import { ApmPluginStartDeps } from '../../../../plugin';
+import { useApmDataView } from '../../../../hooks/use_apm_data_view';
+import { useTheme } from '../../../../hooks/use_theme';
+
+export type OnAddFilter = ({
+  fieldName,
+  fieldValue,
+  include,
+}: {
+  fieldName: string;
+  fieldValue: string | number;
+  include: boolean;
+}) => void;
+
+const defaultFilters: Filter[] = [];
+
+export function FieldStatsPopover({
+  fieldName,
+  fieldValue,
+  onAddFilter,
+}: {
+  fieldName: string;
+  fieldValue: string | number;
+  onAddFilter: OnAddFilter;
+}) {
+  const {
+    query: { kuery: kueryStr },
+  } = useApmParams('/services/{serviceName}');
+
+  const fieldStatsQuery = {
+    query: kueryStr,
+    language: 'kuery',
+  };
+  const { start, end } = useFetchParams();
+
+  const {
+    data,
+    core: { uiSettings },
+  } = useApmPluginContext();
+  const { dataView } = useApmDataView();
+  const {
+    services: { fieldFormats, charts },
+  } = useKibana<ApmPluginStartDeps>();
+
+  const fieldStatsServices: Partial<FieldStatsServices> = {
+    uiSettings,
+    dataViews: data.dataViews,
+    data,
+    fieldFormats,
+    charts,
+  };
+
+  const [infoIsOpen, setInfoOpen] = useState(false);
+  const field = dataView?.getFieldByName(fieldName);
+
+  const closePopover = useCallback(() => setInfoOpen(false), []);
+  const theme = useTheme();
+
+  if (!fieldFormats || !charts || !field || !dataView) return null;
+
+  const addFilter = (
+    popoverField: DataViewField | '_exists_',
+    value: unknown,
+    type: '+' | '-'
+  ) => {
+    if (
+      popoverField !== '_exists_' &&
+      (typeof value === 'number' || typeof value === 'string')
+    ) {
+      onAddFilter({
+        fieldName: popoverField.name,
+        fieldValue: value,
+        include: type === '+',
+      });
+    }
+  };
+
+  const trigger = (
+    <EuiToolTip
+      content={i18n.translate(
+        'xpack.apm.correlations.fieldContextPopover.descriptionTooltipContent',
+        {
+          defaultMessage: 'Show top 10 field values',
+        }
+      )}
+    >
+      <EuiButtonIcon
+        iconType="inspect"
+        onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
+          setInfoOpen(!infoIsOpen);
+        }}
+        aria-label={i18n.translate(
+          'xpack.apm.correlations.fieldContextPopover.topFieldValuesAriaLabel',
+          {
+            defaultMessage: 'Show top 10 field values',
+          }
+        )}
+        data-test-subj={'apmCorrelationsContextPopoverButton'}
+        style={{ marginLeft: theme.eui.euiSizeXS }}
+      />
+    </EuiToolTip>
+  );
+
+  const overrideFieldTopValueBar = (
```

Comment:
Would be great to wrap in `useCallback`

Reply:
Updated here https://github.com/elastic/kibana/pull/143416/commits/54769031be5ac55d822d3ce39b7d844cccf52c89